Quaver.Shared/Screens/Edit/UI/AutoMods/EditorAutoModPanelContainer.cs

```diff
@@ -61,7 +62,17 @@ public EditorAutoModPanelContainer(EditScreen screen)
             {
                 if (x == Screen.Map)
                     return;
-
+                try
+                {
+                    File.OpenRead(Directory.GetCurrentDirectory() + "/Songs/" + x.Directory + "/" + x.Path);
+                } catch (FileNotFoundException)
+                {
+                    mapset.Remove(x.LoadQua());
+                    MapDatabaseCache.RemoveMap(x);
+                    MapDatabaseCache.Load(false);
+                    return;
```

Comment:
I don't think the AutoMod should be responsible for removing files/updating the cache. If files aren't able to be read, then the AutoMod should notify the user of the issue.

Reply:
Moved the database update in the EditorScreen file.